pkg/ibmcsidriver/controller.go

```diff
@@ -379,24 +381,18 @@ func (csiCS *CSIControllerServer) ListVolumes(ctx context.Context, req *csi.List
 	// populate requestID in the context
 	ctx = context.WithValue(ctx, provider.RequestID, requestID)
 	ctxLogger.Info("CSIControllerServer-ListVolumes...", zap.Reflect("Request", req))
-	defer metrics.UpdateDurationFromStart(ctxLogger, metrics.FunctionLabel("ListVolumes"), time.Now())
+	defer metrics.UpdateDurationFromStart(ctxLogger, metrics.FunctionLabel("CSIListVolumes"), time.Now())
 
 	session, err := csiCS.CSIProvider.GetProviderSession(ctx, ctxLogger)
 	if err != nil {
-		return nil, commonError.GetCSIError(ctxLogger, commonError.InternalError, requestID, err)
+		return nil, commonError.GetCSIBackendError(ctxLogger, requestID, err)
 	}
 
 	maxEntries := int(req.MaxEntries)
 	tags := map[string]string{}
 	volumeList, err := session.ListVolumes(maxEntries, req.StartingToken, tags)
 	if err != nil {
-		errCode := err.(providerError.Message).Code
-		if strings.Contains(errCode, "InvalidListVolumesLimit") {
-			return nil, commonError.GetCSIError(ctxLogger, commonError.InvalidParameters, requestID, err)
-		} else if strings.Contains(errCode, "StartVolumeIDNotFound") {
-			return nil, commonError.GetCSIError(ctxLogger, commonError.StartVolumeIDNotFound, requestID, err, req.StartingToken)
-		}
-		return nil, commonError.GetCSIError(ctxLogger, commonError.ListVolumesFailed, requestID, err)
```

Comment:
@MeetanshiGupta Only replace the defatult part keep else if as it is , this will break sanity. I had raised another PR in VPC FILE to revert this.
Sample PR : https://github.com/IBM/ibm-vpc-file-csi-driver/pull/113/files